### DISCUSSION TITLE
Mysql restore failed because of corrupted root user password.

### DISCUSSION THREAD

**Discussion by bhavitha8883** [mydumper/mydumper#807] [Q&A]
![image](https://user-images.githubusercontent.com/32682454/186296791-2fed4263-c1e0-4d71-9fb9-f17ff9e7e80e.png)

**Comment by davidducos** [comment]
Hi @bhavitha8883,
you will need to provide a detailed reproducible test case, with the commands executed and the minimal data needed to see the error. BTW, you should be using the latest version of MyDumper.

**Reply by bhavitha8883** [reply]
Hi @davidducos : I did some backups and restores with 0.9.5 and latest please see the table below.

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<div style='direction:ltr'>



Test   case No. | Backup |   |   | Restore |   |   | Result | Error
-- | -- | -- | -- | -- | -- | -- | -- | --
  | Mydumper   Version | Mysql   Version | K8s   Version | Mydumper   Version | Mysql   Version | K8s   Version |   |  
1 | 0.12.5-3 | 8.0.29 | 1.21.10 | 0.12.5-3 | 8.0.29 | 1.21.10 | Successful | -
2 | 0.9.5 | 8.0.29 | 1.21.10 | 0.9.5 | 8.0.29 | 1.21.10 | Unuccessful   But   successful   after   changing    auth   plugin for    root   user | Same   as 6th
3 | 0.9.5 | 8.0.22 | 1.21.10 | 0.12.5-3 | 8.0.29 | 1.21.10 | unsuccessful | backup   is not a mydumper backup
4 | 0.9.5 | 8.0.22 | 1.21.10 | 0.12.5-3 | 8.0.29 | 1.21.10 | unsuccessful | It   was not possible to process file
5 | 0.9.5 | 8.0.22 | 1.19.10 | 0.12.5-3 | 8.0.29 | 1.21.10 | unsuccessful | It   was not possible to process file
6 | 0.9.5 | 8.0.22(1.1.0) | 1.19.10 | 0.9.5 | 8.0.29 | 1.21.10 | unsuccessful   But successful   after changing    auth plugin for    root user | (myloader:28):   CRITICAL **: 12:42:29.545: Error connection to database:    Plugin   caching_sha2_password could not be loaded:    /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so:      cannot   open shared object file: No such file or directory



</div>


</body>

</html>


So, 3,4, and 5 proves that as you mentioned here (https://github.com/mydumper/mydumper/issues/598 ) we could able to restore a backup which made from different version of the dumper.

And If you see the 2 and 6 that is what my problem, Im using the 0.9.5 for both backup and restore on a fresh mysql instance regardless of the k8s/mysql version it is throwing the error as mentioned above.

Could you please suggest me here how to get rid of this issue.(as a work-around i altered the root user auth plugin to mysql_native).

Thank you so much in advance.

**Reply by davidducos** [reply]
Why don't you use v0.12.5-3 for backup and restore?

**Reply by bhavitha8883** [reply]
Yes, we can use it and make backups with latest version but there are some setbacks for us with infra versions of our clusters. But it would be great if we know the solution for above issue. If not somehow we have to patch our older infra versions.

Thank you.
